x-pack/functionbeat/provider/gcp/gcp/pubsub.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/feature"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/x-pack/functionbeat/function/core"
+	"github.com/elastic/beats/x-pack/functionbeat/function/provider"
+	"github.com/elastic/beats/x-pack/functionbeat/provider/gcp/gcp/transformer"
+)
+
+// PubSub represents a Google Cloud function which reads event from Google Pub/Sub triggers.
+type PubSub struct {
+	log    *logp.Logger
+	config *FunctionConfig
+}
+
+// PubSubMsg is an alias to string
```

Comment:
PubSubMsg is used to set/get pub-sub related fields in a context.Context.

Better not use a string, but introduce getter/setter for some type safety (plus reduce the chance of typos). For example see godoc comment on `(context.Context).Value`.